x-pack/plugins/observability_solution/observability_ai_assistant/server/service/index.ts

```diff
@@ -339,6 +339,7 @@ export class ObservabilityAIAssistantService {
         this.kbService!.queue(
           entries.flatMap((entry) => {
             const entryWithSystemProperties = {
+              type: 'contextual' as const,
               ...entry,
               '@timestamp': new Date().toISOString(),
               doc_id: entry.id,
```

Comment:
I'm not sure why the properties below like `public`, `confidence` and `is_correction` are declared after `...entry` thus overwriting the values in the entry instead of acting as a default.

Reply:
This part still confuses me. We have two ways to add things to KB: `queue` and `addEntry`.

Afaict `queue` is only called from `addToKnowledgeBase`, which is only called from `addCategoryToKnowledgeBase`. So the queueing is solely for adding category entries?

`addEntry` is called from `createKnowledgeBaseEntry` which is the one we called from the API (`POST /internal/observability_ai_assistant/kb/entries/save`)

Reply:
Yes we load some docs on startup, so that needs a queuing system because the model might not be ready at that point.

Reply:
Yeah, I still think instead of a queue we could have a promise that resolves when the kb is ready:

```ts
  await kbService.waitForKnowledgeBaseToBeReady();
  return kbService.addEntry(entry);
```

We could still encapsulate this in a task so it's only picked up by a single instance.

Reply:
(not that it's a concern I want to address in this PR)